cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -72,6 +76,10 @@
   // set of (EndPoint, Field)s which were generated by it.
   private Map<EndPointField, Set<EndPointField>> outgoingSummary;
 
+  // We maintain the map of fields dropped in this lineage; ones with incoming, but no outgoing.
+  // Also the TransformOperation that dropped that field.
+  private final Map<String, TransformOperation> droppedFields;
```

Comment:
Just maintain a `Set<TransformOperation>` here, you don't need to use the key, the keys are in the inputs of the transform